aten/src/ATen/native/sparse/cuda/SparseCUDATensor.cpp

```diff
@@ -46,7 +46,11 @@ SparseTensor& sparse_mask_out_cuda(SparseTensor& r, const Tensor& t, const Spars
     view_size[d + 1] = mask.size(mask.sparse_dim() + d);
   }
 
-  Tensor t_view = t.view(view_size);
+  Tensor t_view;
+  if (t.is_contiguous())
+      t_view = t.view(view_size);
+  else
+      t_view = t.contiguous().view(view_size);
```

Comment:
Compared to unconditionally doing the `.contiguous()`, this saves only the dispatch overhead right?
Or is `.contiguous()` actually more expensive than dispatch + `.is_contiguous()`?

Reply:
Right, this saves only dispatch overhead. No-op `contiguous` is ~100 ns (because it goes through dispatch), `is_contiguous()` is ~2 ns.

Reply:
Ok !